sdk/Lusid.Sdk.Tests/tutorials/Ibor/StructuredResultStore.cs

```diff
@@ -0,0 +1,946 @@
+            
+            // Create result data key rule specifying 
+            string resourceKey = "UnitResult/*";
+            var resultDataKeyRule = new ResultDataKeyRule("Client", documentScope, documentId, resourceKey: resourceKey, documentResultType: resultType, resultKeyRuleType: ResultKeyRule.ResultKeyRuleTypeEnum.ResultDataKeyRule, quoteInterval: "1D");
+
+            // Create and upsert a recipe with the result data key rule
+            var pricingOptions = new PricingOptions {AllowAnyInstrumentsWithSecUidToPriceOffLookup = false, AllowPartiallySuccessfulEvaluation = true};
+            var pricingContext = new PricingContext(null, null, pricingOptions, new List<ResultKeyRule>{resultDataKeyRule} );
+            var configurationRecipe = new ConfigurationRecipe(documentScope, "recipe", new MarketContext(), pricingContext);
+            var upsertRecipeRequest = new UpsertRecipeRequest(configurationRecipe, null);
+            _recipeApi.UpsertConfigurationRecipe(upsertRecipeRequest);
+            
+            // Create a valuation request, requesting multiple results including Strategy and Country.
+            // Applying this request to two portfolios.
+            // Sorting by portfolio id for reliable reproducibility.
+            var valuationRequest = new ValuationRequest(
+                recipeId: new ResourceId(documentScope, "recipe"),
+                metrics: new List<AggregateSpec>
+                {
+                    new AggregateSpec(TestDataUtilities.Luid, AggregateSpec.OpEnum.Value),
+                    new AggregateSpec("Portfolio/Id", AggregateSpec.OpEnum.Value),
+                    new AggregateSpec("Portfolio/Scope", AggregateSpec.OpEnum.Value),
+                    new AggregateSpec($"Transaction/{scope1}/Strategy", AggregateSpec.OpEnum.Value),
+                    new AggregateSpec($"Transaction/{scope2}/Country", AggregateSpec.OpEnum.Value),
+                    new AggregateSpec("UnitResult/Returns/YtD", AggregateSpec.OpEnum.Value),
+                    new AggregateSpec("UnitResult/UserDefinedKey", AggregateSpec.OpEnum.Value),
+                },
+                portfolioEntityIds: new List<PortfolioEntityId>
+                {
+                    new PortfolioEntityId(scope1, portfolioCode1),
+                    new PortfolioEntityId(scope1, portfolioCode2)
+                },
+                valuationSchedule: new ValuationSchedule(effectiveAt: effectiveAt),
+                sort: new List<OrderBySpec>
+                {
+                    new OrderBySpec("Portfolio/Id", OrderBySpec.SortOrderEnum.Ascending)
+                }
+            );
+    
+            // Perform valuation and obtain results
+            var results = _apiFactory.Api<IAggregationApi>().GetValuation(valuationRequest);
+
+            
+            // We expect the following results
+            // | LusidInstrumentId | PortfolioId | PortfolioScope | Strategy  | Country | YtD      | UserDefinedKey |
```

Comment:
I think it would be even more realistic if this included the Holding/Units value (and even the PV if you wanted)

Reply:
I was thinking about this initially but was a bit worried when user looks at rows with LusidInstrumentId `CCY_USD` they may be a bit confused because that instrument has not been created and thus its Units and PV are not known. Will add